GGRC-4874/GGRC-4875 Add spinner for Cycle Tasks actions in TreeView

gel1os opened the pull request:
<!-- If your PR depends on other tickets or PRs, you can list them here
# Dependencies

This PR is `on hold` until the dependencies are merged:

- [ ] GGRC-1234
- [ ] #1234
-->

# Issue description

Display spinner for cycle tasks in TreeView, while a request to the server is in progress.

# Steps to test the changes

1. Create any WF; 
2. In the Setup tab add tasks to Task Group ;
3. Activate WF;
4. In the Active Cycles tab expand TreeView with Cycle Tasks;
5. Click on "Start", "Undo" or "Finish" button;

# Solution description

Modify `cycle-task-actions.mustache` template to include `spinner` component when an action button is disabled.

# Sanity checklist

- [x] I have clicked through the app to make sure my changes work and not break the app.
- [ ] I have applied the correct milestone and labels.
- [x] My changes fix the issue described in the description (and do nothing else). 🤞
- [x] My changes are covered by tests.
- [ ] My changes follow our [performance guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/performance.rst).
- [x] My changes follow our [js](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/javascript.rst) and/or [python](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/python.rst) guidelines.
- [x] My commits follow our [commit guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/git/how_to_write_a_commit_message.rst).

<!-- If your PR includes a migration include the additional checklist items
# Migration checklist
- [ ] Migration passes all checks from our [PR review guidelines](https://github.com/google/ggrc-core/blob/dev/docs/source/contributing/git/reviewing_pull_requests.rst#reviewing-a-pr-containing-database-migration-scripts)
-->

# PR Review checklist

- [x] The changes fix the issue and don't cause any apparent regressions.
- [x] Labels and milestone are correctly set.
- [x] The solution description matches the changes in the code.
- [x] There is no apparent way to improve the performance & design of the new code.
- [x] The pull request is opened against the correct base branch.
- [ ] Upon merging, the Jira ticket's fixversion is correctly set and the ticket is moved to "QA - In Progress".

<!-- If your code is not finished yet can include a TODO check list
# TODO

- [ ] First item on the TODO
-->

gel1os commented:
I signed it!

valchanin commented on src/ggrc-client/js/components/cycle-task-actions/cycle-task-actions.mustache:
```diff
@@ -10,7 +10,10 @@
 <div class="flex-box item-actions{{#if disabled}} disabled{{/if}}">
-  {{#with_review_task}}
+  {{#if disabled}}
+    <spinner toggle="disabled" size="medium"></spinner>
```
Please use can.stache syntax instead of can.mustache - {key}="value" (https://v2.canjs.com/docs/can.Component.html#section_Differencesbetweencomponentsincan_mustacheandcan_stache)

valchanin commented:
P.S. Since you're using spinner component in `cycle-task-actions` you have to import it. Currently it works because spinner already was in the same bundle, but we need to take care about dependencies.

gel1os commented:
@valchanin Hi, Artsiom, I've fixed a `cycle-task-actions` import and used can.stache syntax in `spinner` component.

I've also added an error handling, but there is a minor problem, perhaps you can help me with it:

So I'm testing the error cases by using Chrome's `offline` checkbox (DevTools -> Network Tab). When offline mode it's enabled, all the requests to server fail and the status of cycle task doesn't change if you click on Start / Finish / Undo buttons (which is expected). 

But when I turn the `online` mode afterwards and try to perform the action again, GET request to `api/cycle_task_group_object_tasks` returns 200, however `await instance.refresh();` function throws an exception. As a result a status is not updated. It will be updated correctly after consequent clicks on Start / Finish / Undo buttons.

It would be great if you could give me some advice about what the problem might be. Let me know if you need more information.

AlexNo commented:
@gel1os eslint check has failed.

```
A  Issue count difference: 1 
A  FAIL: The PR would increase the ESLInt issue count. 
```

AlexNo commented on src/ggrc-client/js/components/cycle-task-actions/cycle-task-actions.mustache:
```diff
@@ -10,7 +10,10 @@
 {{#if_instance_of instance 'CycleTaskGroupObjectTask'}}
 <div class="flex-box item-actions{{#if disabled}} disabled{{/if}}">
-  {{#with_review_task}}
+  {{#if disabled}}
```
I'd suggest to hide buttons use css instead of remove them from the DOM. Since we already have `disabled` class on the parent element

gel1os commented:
@AlexNo Thanks, it's fixed.
